### PR TITLE
add: marshal,unmarshal added test for check values symmetry

### PR DESCRIPTION
**Pull request by illia-li**
Added test for check symmetry of marshaling, unmarshalling.
As part of the solution for #237

### CODE REVIEW

File: symmetry_values_test.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+	"net"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+)
+
+const testProto = 2
+
+func TestSymmetry(t *testing.T) {
```
Comment (dkropachev) [inline symmetry_values_test.go:17]: There is two problems with this implementations:
1. It does not actually check marshal correctness, meaning that it does not check that certain structure end up marshaled into certain data
2. It is so hard to understand what exactly is tested that you need to debug it to see it.

To address that please redo tests in the following format, don't replicate it `100%`, use it just as point of reference:
```
type marshalTypeTestCase struct {
	name          string
	cqlType       TypeInfo
	value         interface{}
	expected      []byte
	skipUnmarshal string
}

var marshalTestCases = []marshalTypeTestCase{}

func TestMarshal(t *testing.T) {
	for id := range marshalTestCases {
		tcase := marshalTestCases[id]
		tcase.cqlType.Version()
		t.Run(fmt.Sprint("%s-%d-%s", tcase.cqlType.Type().String(), tcase.cqlType.Version(), tcase.name), func(t *testing.T) {
			data, err := Marshal(tcase.cqlType, tcase.value)
			if err != nil {
				t.Fatal("failed to marshal value", err)
			}
			if !reflect.DeepEqual(data, tcase.expected) {
				t.Fatalf("expected %v, got %v", tcase.expected, data)
			}

			gotUnmarshal := reflect.New(reflect.TypeOf(tcase.value))
			err = Unmarshal(tcase.cqlType, data, gotUnmarshal)
			if err != nil {
				t.Fatal("failed to unmarshal to the same type", err)
			}

			if tcase.skipUnmarshal != "" {
				t.Skip(tcase.skipUnmarshal)
				return
			}

			if !reflect.DeepEqual(tcase.value, gotUnmarshal) {
				t.Fatalf("expected %v, got %v", tcase.expected, data)
			}
		})
	}
}
```

Also please reconcile with what already exists in `marshal_test.go` and remove old duplicates.

File: symmetry_values_test.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
+package gocql
```
Comment (dkropachev) [inline symmetry_values_test.go:1]: Please move it to `marshal_test.go`

File: go.mod [hunk 3]
```diff
@@ -5,6 +5,7 @@ require (
 	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869 // indirect
 	github.com/golang/snappy v0.0.3
 	github.com/google/go-cmp v0.4.0
+	github.com/google/uuid v1.6.0
```
Comment (dkropachev) [inline go.mod:8]: Please don't bring new dependancy without real necessity, in this case it is not realy needed.

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: symmetry_values_test.go (hunk 1 above)
Comment (illia-li) [reply symmetry_values_test.go]: Fixed

File: symmetry_values_test.go (hunk 2 above)
Comment (illia-li) [reply symmetry_values_test.go]: Fixed

File: go.mod (hunk 3 above)
Comment (illia-li) [reply go.mod:8]: Fixed

**Comment by illia-li** [comment]
Not actual

**Pull request closed without merging**